src/langchain_google_alloydb_pg/vectorstore.py

```diff
@@ -711,17 +882,170 @@ async def asimilarity_search_with_score_by_vector(
+
+    def _similarity_search_with_relevance_scores(
+        self,
+        query: Optional[str] = None,
+        image_uri: Optional[str] = None,
+        k: int = 4,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """
+        Return docs and relevance scores in the range [0, 1].
+
+        0 is dissimilar, 1 is most similar.
+
+        Args:
+            query: Input text.
+            k: Number of Documents to return. Defaults to 4.
+            **kwargs: kwargs to be passed to similarity search. Should include:
+                score_threshold: Optional, a floating point value between 0 to 1 to
+                    filter the resulting set of retrieved docs
+
+        Returns:
+            List of Tuples of (doc, similarity_score)
+        """
+        relevance_score_fn = self._select_relevance_score_fn()
+        docs_and_scores = self.similarity_search_with_score(
+            query=query, image_uri=image_uri, k=k, **kwargs
+        )
+        return [(doc, relevance_score_fn(score)) for doc, score in docs_and_scores]
+
+    def similarity_search_with_relevance_scores(
+        self,
+        query: Optional[str] = None,
+        image_uri: Optional[str] = None,
+        k: int = 4,
+        **kwargs: Any,
+    ) -> List[Tuple[Document, float]]:
+        """Return docs and relevance scores in the range [0, 1].
+
+        0 is dissimilar, 1 is most similar.
+
+        Args:
+            query: Input text.
+            k: Number of Documents to return. Defaults to 4.
+            **kwargs: kwargs to be passed to similarity search. Should include:
+                score_threshold: Optional, a floating point value between 0 to 1 to
+                    filter the resulting set of retrieved docs.
+
+        Returns:
+            List of Tuples of (doc, similarity_score).
+        """
+        score_threshold = kwargs.pop("score_threshold", None)
+
+        docs_and_similarities = self._similarity_search_with_relevance_scores(
+            query=query, image_uri=image_uri, k=k, **kwargs
+        )
+        if any(
+            similarity < 0.0 or similarity > 1.0
+            for _, similarity in docs_and_similarities
+        ):
+            warnings.warn(
+                "Relevance scores must be between"
+                f" 0 and 1, got {docs_and_similarities}"
+            )
+
+        if score_threshold is not None:
+            docs_and_similarities = [
+                (doc, similarity)
+                for doc, similarity in docs_and_similarities
+                if similarity >= score_threshold
+            ]
+            if len(docs_and_similarities) == 0:
+                warnings.warn(
+                    "No relevant docs were retrieved using the relevance score"
+                    f" threshold {score_threshold}"
+                )
+        return docs_and_similarities
+
+    async def asimilarity_search_with_relevance_scores(
```

Comment:
These functions don't take image_uri as inputs so it raises error for both the image and text search because the LangChain implementation doesn't specify input parameter name when calling corresponding functions (so it misunderstands kwargs as image_uri input which raised error). This is why I modify it to take image_uri and added parameter specification for this function.